src/Generator/Caller/SimpleCaller.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+/*
+ * This file is part of the Alice package.
+ *
+ * (c) Nelmio <hello@nelm.io>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Nelmio\Alice\Generator\Caller;
+
+use Nelmio\Alice\Definition\MethodCallInterface;
+use Nelmio\Alice\Definition\ValueInterface;
+use Nelmio\Alice\FixtureInterface;
+use Nelmio\Alice\Generator\CallerInterface;
+use Nelmio\Alice\Generator\GenerationContext;
+use Nelmio\Alice\Generator\ResolvedFixtureSet;
+use Nelmio\Alice\Generator\ValueResolverAwareInterface;
+use Nelmio\Alice\Generator\ValueResolverInterface;
+use Nelmio\Alice\ObjectInterface;
+use Nelmio\Alice\Throwable\Exception\Generator\Resolver\ResolverNotFoundExceptionFactory;
+use Nelmio\Alice\Throwable\Exception\Generator\Resolver\UnresolvableValueDuringGenerationExceptionFactory;
+use Nelmio\Alice\Throwable\ResolutionThrowable;
+
+final class SimpleCaller implements CallerInterface, ValueResolverAwareInterface
+{
+    /**
+     * @var ValueResolverInterface
+     */
+    private $resolver;
+
+    public function __construct(ValueResolverInterface $resolver = null)
+    {
+        $this->resolver = $resolver;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function withValueResolver(ValueResolverInterface $resolver): self
+    {
+        return new self($resolver);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function doCallsOn(
+        ObjectInterface $object,
+        ResolvedFixtureSet $fixtureSet,
+        GenerationContext $context
+    ): ResolvedFixtureSet {
+        if (null === $this->resolver) {
+            throw ResolverNotFoundExceptionFactory::createUnexpectedCall(__METHOD__);
+        }
+
+        $fixture = $fixtureSet->getFixtures()->get($object->getId());
+        $calls = $fixture->getSpecs()->getMethodCalls();
+
+        $scope = [
```

Comment:
Ah right it's needed for the `EvaluatedValueResolver`. Could you please add a test case for that in the integration test to not forget about it? Something like this should trigger it:

```yaml
Dummy:
  dummy1: {}
  dummy2:
    __calls:
      - setRelatedDummy: [ '<(@dummy1)>' ]
```